README.md

```diff
@@ -5,57 +5,166 @@
-
-
-Note: for faster gradle operations, add `org.gradle.daemon=true` to your `~/.gradle/gradle.properties` file.  This will keep a gradle daemon running in the background and avoid the ~6s gradle start up time on every command.  
+* Git 2.5 or greater
+* Optional, but recommended:
+    * Python 2.6 or greater (needed for running the `gatk-launch` frontend script)
+    * Gradle 2.7 or greater (recommended, but can use the `./gradlew` script instead which will download an appropriate gradle version for you)
+    * R 3.1.3 (needed for producing plots in certain tools, and for running the test suite)
+    * [git-lfs](https://git-lfs.github.com/) 1.1.0 or greater (needed to download large files for the complete test suite).
+      Run `git lfs install` after downloading, followed by `git lfs pull` to download the large files. The download is ~500 MB.
+
+##Quick Start Guide
+
+* Build the GATK: `gradle installAll`
+* Get a list of available tools: `./gatk-launch --help`
+* Run a tool: `./gatk-launch PrintReads -I input.bam -O output.bam`
+* Get help on a particular tool: `./gatk-launch ToolName --help`
+
+##Building GATK4
+
+* To build everything you need to run all GATK tools, run **`gradle installAll`**.
+    * This does both the standard build and the Spark build described below, and is the recommended way to compile the GATK.
+
+* To do a standard build only to save time, run **`gradle installDist`**.
+    * The resulting executable will be `build/install/gatk/bin/gatk`,
+      and will be suitable for running non-Spark tools and for running Spark tools locally (not on a cluster).
+    * This executable is **not** a fully-packaged executable that can be copied across machines, it's just
+      for running in-place. To create a packaged version of this build, run `gradle shadowJar`. The
+      packaged jar will be in `build/libs/` with a name like `gatk-all-*-SNAPSHOT-shadowJar.jar`.
+
+* To just build a special jar suitable for running spark tools on a cluster, run **`gradle installSpark`**.
+    * The resulting jar will be in `build/libs/` and will have a name like `gatk-all-*-SNAPSHOT-spark.jar`
+    * This jar will not include Spark and Hadoop libraries, in order to allow
+      the versions of Spark and Hadoop installed on your cluster to be used.
+    * If you've previously built a Spark jar, you'll need to run `gradle clean` first to remove old versions
+      of the jar from `build/libs`, otherwise `gatk-launch` will not know which jar to use.
+
+* To run all tests, run **`gradle test`**.
+    * Test report is in `build/reports/tests/index.html`.
+    * What will happen depends on the value of the `CLOUD` environment variable: if it's `false` or
+      unset then only local tests are run, if it's `mandatory` then it'll run only the cloud tests,
```

Comment:
Should mention that cloud tests require being logged into gcloud and authenticated with a project that has access to the test data.